debootstrap uses absolute paths for a few binaries, which means it tries to use
the host versions instead of the Nix-provided versions when run under a nix
shell.  This fails under fakeroot due to the binaries' libc not matching the
libc used by the Nix-provided fakeroot.

--- a/debootstrap
+++ b/debootstrap
@@ -500,10 +500,10 @@
 
 if in_path dpkg && \
      dpkg --print-architecture >/dev/null 2>&1; then
-	HOST_ARCH=$(/usr/bin/dpkg --print-architecture)
+	HOST_ARCH=$(dpkg --print-architecture)
 elif in_path udpkg && \
      udpkg --print-architecture >/dev/null 2>&1; then
-	HOST_ARCH=$(/usr/bin/udpkg --print-architecture)
+	HOST_ARCH=$(udpkg --print-architecture)
 elif [ -e "$DEBOOTSTRAP_DIR/arch" ]; then
 	HOST_ARCH=$(cat "$DEBOOTSTRAP_DIR/arch")
 fi
@@ -611,7 +611,7 @@
 
 ###########################################################################
 
-if [ -x /usr/bin/arch-test ] && am_doing_phase second_stage; then
+if in_path arch-test && am_doing_phase second_stage; then
 	if doing_variant fakechroot; then
 		ret=0; arch-test "$ARCH" || ret=$?
 	# Avoid failure with old arch-test package
